tests/dbz-twin/rac/db-check.py

```
@@ -0,0 +1,383 @@
+
+    conn = oracledb.connect(dsn)
+    cursor = conn.cursor()
+    state = {}
+
+    for table_name, pk_col, compare_cols in TABLES:
+        all_cols = list(dict.fromkeys([pk_col, 'EVENT_ID'] + compare_cols))
+        col_list = ', '.join(all_cols)
+        try:
+            cursor.execute(
+                f"SELECT {col_list} FROM olr_test.{table_name} "
+                f"WHERE EVENT_ID NOT IN ('SEED', 'SENTINEL')"
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)